nexus_constructor/instrument_zooming_3d_window.py

```diff
@@ -0,0 +1,33 @@
+from PySide2 import QtGui
+from PySide2.Qt3DExtras import Qt3DExtras
+from PySide2.QtCore import Qt
+
+
+class InstrumentZooming3DWindow(Qt3DExtras.Qt3DWindow):
+    def __init__(self):
+        """
+        A custom 3D window that only zooms in on the instrument components when the escape key is pressed.
+        """
+        super().__init__()
+        self.component_root_entity = None
+        self.prev_position = None
+
+    def set_component_root_entity(self, component_root_entity):
```

Comment:
You should pass this entity in as part of the constructor. Otherwise you risk calling `viewEntity(None)` below.